client/src/main/java/io/split/client/LocalhostSplitFile.java

```diff
@@ -0,0 +1,130 @@
+        Preconditions.checkNotNull(fileName);
+
+        _splitFactory = Preconditions.checkNotNull(splitFactory);
+        _file = new File(directory, fileName);
+        _watcher = FileSystems.getDefault().newWatchService();
+        _stop = new AtomicBoolean(false);
+    }
+
+    private boolean isStopped() {
+        return _stop.get();
+    }
+
+    private void stopThread() {
+        _stop.set(true);
+    }
+
+    public void registerWatcher() throws IOException {
+        Path path = _file.toPath().toAbsolutePath().getParent();
+        path.register(_watcher, new WatchEvent.Kind[]{StandardWatchEventKinds.ENTRY_MODIFY}, SensitivityWatchEventModifier.HIGH);
+    }
+
+    @Override
+    public void run() {
+        try {
+            while (!isStopped()) {
+                WatchKey key;
+                try {
+                    key = _watcher.poll(250, TimeUnit.MILLISECONDS);
+                } catch (InterruptedException e) {
+                    stopThread();
+                    return;
+                }
+                if (key == null) {
+                    Thread.yield();
+                    continue;
+                }
+
+                for (WatchEvent<?> event : key.pollEvents()) {
+                    WatchEvent.Kind<?> kind = event.kind();
+
+                    @SuppressWarnings("unchecked")
+                    WatchEvent<Path> ev = (WatchEvent<Path>) event;
+                    Path filename = ev.context();
+
+                    if (kind == StandardWatchEventKinds.OVERFLOW) {
+                        Thread.yield();
+                        continue;
+                    } else if (kind == java.nio.file.StandardWatchEventKinds.ENTRY_MODIFY
+                            && filename.toString().equals(_file.getName())) {
+                        Map<String, String> featureToSplitMap = readOnSplits();
+                        _splitFactory.updateFeatureToTreatmentMap(featureToSplitMap);
+                        _log.info("Detected change in Local Splits file - Splits Reloaded! file={}", _file.getPath());
+                    }
+                    boolean valid = key.reset();
+                    if (!valid) {
+                        break;
+                    }
+                }
+                Thread.yield();
+            }
+        } catch (IOException e) {
+            _log.error("Error reading file: path={}", _file.getPath(), e);
+            stopThread();
+        }
+    }
+
+    public Map<String, String> readOnSplits() throws IOException {
+        Map<String, String> onSplits = Maps.newHashMap();
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(_file))) {
+            for (String line = reader.readLine(); line != null; line = reader.readLine()) {
+                line = line.trim();
+                if (line.isEmpty() || line.startsWith("#")) {
+                    continue;
+                }
+
+                String[] feature_treatment = line.split("\\s+");
```

Comment:
just a personal preference, I would have a POJO with fields feature, treatment rather than this array, but it was already there, so up to you.